numpy/_core/src/multiarray/arraytypes.c.src

```diff
@@ -368,6 +381,9 @@ NPY_NO_EXPORT int
         if (@TYPE@_safe_pyint_setitem(op, &temp) < 0) {
             return -1;
         }
```

Comment:
Maybe easier to just duplicate this call so that we don't need `arg_converted`?  Unfortunately, the error return is leaking the `op` reference.